Python/ceval.c

```diff
@@ -1587,41 +1649,48 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, InterpreterFrame *frame, int thr
     frame->previous = prev_cframe->current_frame;
     cframe.current_frame = frame;
 
+    /* support for generator.throw() */
+    if (throwflag) {
+        if (_Py_EnterRecursiveCall(tstate, "")) {
+            tstate->recursion_remaining--;
+            goto exit_unwind;
+        }
+        TRACE_FUNCTION_THROW_ENTRY();
+        DTRACE_FUNCTION_ENTRY();
+        goto resume_with_error;
+    }
```

Comment:
I want to remove the code for handling `throw()` out of the main dispatch block.
It is an uncommon path, and we will probably want to trace it differently from normal execution.
To avoid duplicating the recursion check, the code after `start_frame` would need to check for `throwflag`, but in the common case that check is unnecessary.